### PR TITLE
Scaling of numerical parameter in contact mechanics

### PR DESCRIPTION
**Pull request by keileg**
## Proposed changes
Main changes:
* Scaled the numerical parameter in contact mechanics with the material properties of the domain.
* Bugfix in fractured meshing, relating to large fractures (rounding error).

I did some design choices relating to the contact mechanics. Please review these critically.

Do we need tests? Answer may be yes.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [x] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).

### CODE REVIEW

File: src/porepy/fracs/structured.py [hunk 1]
```diff
@@ -399,27 +399,42 @@ def _create_lower_dim_grids_2d(g_2d, fracs, nx):
...
+    # yet. Find this by identifying the coordinate axis which is numerical zero. This
+    # requires a scaling with the fracture size to avoid numerical issues. Find the size
+    # of the fracture by first finding the bounding box of the points.
+    bounding_box: dict[str, float] = pp.domain.bounding_box_of_point_cloud(loc_coord)
+    # The bounding box has keys 'xmin', 'xmax', 'ymin', 'ymax', 'zmin', 'zmax'. The list
+    # comprehension below loops over the min and max values of each coordinate and finds
+    # the largest difference.
+    fracture_size = np.max(
+        [bounding_box[f"{s}max"] - bounding_box[f"{s}min"] for s in "xyz"]
+    )
```
Comment (IvarStefansson) [inline src/porepy/fracs/structured.py:424]: Nice!

File: src/porepy/models/momentum_balance.py [hunk 2]
```diff
@@ -757,24 +763,32 @@ def contact_mechanics_numerical_constant(
...
+        # for further scaling.
+        shear_modulus = self.solid.shear_modulus()
+        aperture = self.solid.residual_aperture()
+
+        # EK: The scaling factor should not be too large, otherwise the contact problem
+        # may be discretized wrongly. I therefore introduce a safety factor here; its
+        # value is somewhat arbitrary.
+        safety_factor = 1e-1
+
+        val = safety_factor * shear_modulus * aperture
```
Comment (IvarStefansson) [inline src/porepy/models/momentum_balance.py:790]: Why do you want to scale with aperture? If anything, I'd expect inverse length scaling.

Comment (IvarStefansson) [reply src/porepy/models/momentum_balance.py:790]: But this coefficient is multiplied with the jump, isn't it? Shall we discuss in person?

Comment (keileg) [reply src/porepy/models/momentum_balance.py:790]: You are right. I'll remove the aperture.

**Review by IvarStefansson — APPROVED** [review]

**Pull request merged**